x/genutil/legacy/v043/migrate.go

```diff
@@ -0,0 +1,27 @@
+package v043
+
+import (
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/x/genutil/types"
+	v040gov "github.com/cosmos/cosmos-sdk/x/gov/legacy/v040"
+	v043gov "github.com/cosmos/cosmos-sdk/x/gov/legacy/v043"
+)
+
+// Migrate migrates exported state from v0.40 to a v0.43 genesis state.
+func Migrate(appState types.AppMap, clientCtx client.Context) types.AppMap {
+	// Migrate x/gov.
+	if appState[v040gov.ModuleName] != nil {
+		// unmarshal relative source genesis application state
+		var oldGovState v040gov.GenesisState
```

Comment:
Why do we store/copy genesis in a state?

Reply:
I'm not sure I understand.

`GenesisState` means "the state of the genesis", it's a struct. We have InitGenesis/ExportGenesis functions that convert actual store to/from GenesisState structs.

Reply:
Ah, this is not the _on fly migration_?

Reply:
yeah, it's JSON. I renamed ths PR title